src/services/authenticationService.ts

```diff
@@ -16,5 +19,18 @@ export const authenticate = async (connection: IBasicConnection): Promise<void>
       default:
         return Promise.reject(Error(`No supported authentication provider for type: ${connection?.type}`));
     }
-  });
-};
+  })
+}
+
+const navigateToPinCodeForVerification = (): Promise<void> => {
```

Comment:
rename to enterPinCode

And

this can be rewritten to be much cleaner.

const navigateToPinCodeForVerification = aysnc (): Promise<void> => {
	const onVerificationSuccess = async (): Promise<void> => {
      resolve();
    }
    RootNavigation.navigate(ScreenRoutesEnum.LOCK, {
      onVerificationSuccess,
      headerTitle: translate('authenticate_screen_title'),
      headerSubTitle: translate('authenticate_screen_subtitle')
    })
}

Reply:
btw, the idea is to let onVerificationSuccess actually do something. i'm also fine leaving it like this and i will refactor it further when implementing the share flow